src/linear_algebra/basis.lean

```diff
@@ -101,6 +101,14 @@ linear_independent_iff.trans
 λ hf l hl, finsupp.ext $ λ i, classical.by_contradiction $ λ hni, hni $ hf _ _ hl _ $
   finsupp.mem_support_iff.2 hni⟩
 
+theorem linear_independent_iff'' :
+  linear_independent R v ↔ ∀ (s : finset ι) (g : ι → R) (hg : ∀ i ∉ s, g i = 0),
+    ∑ i in s, g i • v i = 0 → ∀ i ∈ s, g i = 0 :=
```

Comment:
```suggestion
    ∑ i in s, g i • v i = 0 → ∀ i, g i = 0 :=
```
You can now strengthen the conclusion, right? Or do you have a strong reason not to do that?